# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/ipc/GamepadEventChannelParent.h
# Commit: d31313f495aa
# Full Hash: d31313f495aa20866d041d044cca726b570f392c
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2021-01-05 04:31:31
# Regressor Bug: 1657404
# File Overlap Count: 2
# Description:
#   Bug 1657404 - Add diag asserts to catch potential double IPDL shutdown r=handyman
#   
#   Based on evidence from crash reports in Bug 1682589, there may be a bug where
#   IPDL intermittently calls `ActorDestroy()` twice on GamepadEventChannelParent,
#   leading to a crash.
# ==============================================================================

diff -r 585c85986b92 -r d31313f495aa dom/gamepad/ipc/GamepadEventChannelParent.h
--- a/dom/gamepad/ipc/GamepadEventChannelParent.h	Mon Jan 04 20:57:43 2021 +0000
+++ b/dom/gamepad/ipc/GamepadEventChannelParent.h	Mon Jan 04 20:38:39 2021 +0000
@@ -41,6 +41,7 @@
   GamepadEventChannelParent();
   ~GamepadEventChannelParent() = default;
 
+  bool mIsShutdown;
   nsCOMPtr<nsIEventTarget> mBackgroundEventTarget;
 };
 
